### PR TITLE
Add ability to filter the container list

### LINKED ISSUE #1103: UI Enhancement: Scrollable container list with search bar
After about 20 containers it becomes unwieldy. It would be good if the container list scrolled so that it had a maximum height and above the container box was a search box that you can type and it would filter the displayed list based on what was typed.

It would make utilizing many containers much more user friendly.

### PR DESCRIPTION
**Pull request by dannycolin**
Resolves #1103 

Preview:

mac-filter-container-list

### CODE REVIEW

File: src/js/popup.js [hunk 1]
```diff
@@ -465,6 +464,24 @@ const Logic = {
...
+    const pattern = /^\s+|\s+$/g
+    const list = Array.from(document.querySelectorAll("#identities-list tr"));
+    const search = document.querySelector("#search-terms").value.replace(pattern, '').toLowerCase();
+
+    for (let i in list) {
+      let text = list[i].querySelector("td div span");
+
+      if (text.innerText.replace(pattern, '').toLowerCase().includes(search) ||
+          search == "" ||
+          search == null) {
```
Comment (stoically) [inline src/js/popup.js:479]:
```suggestion
      if (text.innerText.replace(pattern, '').toLowerCase().includes(search) || !search) {
```

**Review by stoically — APPROVED** [review]

File: src/popup.html [hunk 2]
```diff
@@ -158,6 +158,18 @@ <h4 class="moz-vpn-logotype vpn-status-container-list display-none">Mozilla VPN
       </h4>
     </div>
     <div class="scrollable identities-list">
+      <div class="searchbar">
+        <label for="search-terms"
+               class="hide-label"
+               data-i18n-message-id="filterInputLabel">
```
Comment (flodolo) [inline src/popup.html:164]: I don't see this (`Filter container list by name:`) in the GIF. How is this used?

Comment (dannycolin) [reply src/popup.html:164]: The label is for screen reader only.

**Review by groovecoder — APPROVED** [review]
Code changes look good.

What if we make the pop-up auto-focus into this new container filter list? Or maybe so we don't break existing habits where people might immediately type "1" or "2" to open a container tab in the 1st or 2nd container ... can we make the filter box completely invisible? So, if I open the pop-up and start typing "pers" or "shop" (basically any character that's NOT a number), it will call `filterContainerList` with what I've typed so far? And then, for mega bonus cherries, if the filtered list contains only 1 item, we auto-focus it, so pressing "enter/return" will open it.

I guess I'm thinking of the auto-filter/search experience in other apps:

1. Press Command + K
2. Start typing anything
3. The list of matching "things" gets smaller and smaller until there's just 1 item
4. Press "Return/Enter" to use the 1 item

**Comment by dannycolin** [comment]
> @groovecoder What if we make the pop-up auto-focus into this new container filter list? Or maybe so we don't break existing habits where people might immediately type "1" or "2" to open a container tab in the 1st or 2nd container

I decided not to autofocus exactly because of the hidden functionality to open a container. 

> Can we make the filter box completely invisible? So, if I open the pop-up and start typing "pers" or "shop" (basically any character that's NOT a number), it will call `filterContainerList` with what I've typed so far? 

On a UX point of view, I'd be against hiding the input. I'm already not a fan of the hidden "press a number" function.

> And then, for mega bonus cherries, if the filtered list contains only 1 item, we auto-focus it, so pressing "enter/return" will open it.

Actually, I want to implement something similar in a follow-up PR. Pressing Enter could open whatever the first element in the filtered list is. That first element should always be hightlighted and the starting point if you tab/arrow down so that would be easy to select other items in the list.

However, all of this definitely needs UX work first so I'll merge this PR as-is and we can work on more advanced features in a follow up bug.

**Comment by groovecoder** [comment]
Sounds good, thanks @dannycolin !

**Comment by groovecoder** [comment]
Last idea-bomb: Maybe if someone types "f" for "filter" in the pop-up, that will auto-focus into the container filter input?

**Pull request merged**